crates/oxc_query/src/adapter.rs

```diff
@@ -0,0 +1,436 @@
+use std::{
+    rc::Rc,
+    sync::{Arc, OnceLock},
+};
+
+use oxc_semantic::Semantic;
+use trustfall::{
+    provider::{
+        resolve_coercion_using_schema, resolve_property_with, ContextIterator,
+        ContextOutcomeIterator, EdgeParameters, ResolveEdgeInfo, ResolveInfo, Typename,
+        VertexIterator,
+    },
+    FieldValue, Schema,
+};
+
+use super::vertex::Vertex;
+
+static SCHEMA: OnceLock<Schema> = OnceLock::new();
+
+pub struct Adapter<'a> {
+    pub semantic: Rc<Semantic<'a>>,
+    pub path_components: Vec<Option<String>>,
+}
+
+const SCHEMA_TEXT: &str = include_str!("./schema.graphql");
+
+#[allow(clippy::missing_panics_doc)]
```

Comment:
I don't love that we ignore clippy here, is it worth adding a panic doc comment regardless of whether it will ever happen?

Reply:
It's worth adding the comment.

Reply:
Added in [8984505](https://github.com/web-infra-dev/oxc/pull/599/commits/8984505a9fdb5cb24f11c569e61121f2c649ad4a)